src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -354,7 +366,7 @@ event::ptr primitive_inst::execute(const std::vector<event::ptr>& events) {
 
             _outputs[0] = outputs.at(last_prim_id).get_memory();
 
-            _impl_params->output_layouts[0] = _outputs[0]->get_layout();
+            _impl_params->output_layouts[0] = subgraph->get_output_layout(last_prim_id);
```

Comment:
Why memory layout doesn't work here?

Reply:
In subgraph, memory layout can be allocated as fake shape. 
The policy of the fake alignment is : to store actual layout to be used through shape propagation in impl_params->output_layouts, and allocate output_memory as much as fake aligned kernel needs. 
If we set output layout with fake aligned shape here, it will be recognized as "update_shape" to be true at the next iteration even though the output is not changed.

Reply:
Sorry I forgot to unset enable auto merge ...